### PR TITLE
cctools, apple libiconv: Apple clean up for iOS cross

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

One one hand, this is an "incomplete" mass-rebuild as more changes are needed. OTOH it's nice to upstream things as soon as possible before I forget. LMK what you all think.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @ElvishJerricco

### CODE REVIEW

File: pkgs/os-specific/darwin/apple-source-releases/libiconv/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
-{ stdenv, appleDerivation }:
+{ stdenv, appleDerivation, autoreconfHook }:
```
Comment (Ericson2314) [inline pkgs/os-specific/darwin/apple-source-releases/libiconv/default.nix]: I haven't yet been able to get `autoreconfHook` to work :/, but I think it's needed as the `...-ios` triples are new.

File: pkgs/os-specific/darwin/apple-source-releases/libiconv/default.nix [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 appleDerivation {
-  preConfigure = "cd libiconv";
+  postPatch = "cd libiconv";
```
Comment (matthewbauer) [inline pkgs/os-specific/darwin/apple-source-releases/libiconv/default.nix]: Probably safer to do this:

postUnpack = "sourceRoot=$sourceRoot/libiconv";

**Review by matthewbauer — APPROVED** [review]
Looks good! I was actually just looking at ccports because I couldn't find the man page for otool and found https://github.com/tpoechtrager/cctools-port/issues/40.

**Comment by Ericson2314** [comment]
@matthewbauer fair point; that worked. I really need to head home but anyone feel free to merge this after ofborg then.

**Comment by Ericson2314** [comment]
I built it locally so whatever re the time out.

**Pull request merged**